### ISSUE TITLE
Доступность виджетов при смене шаблона

### ISSUE THREAD

**Issue by zau4man** [instantsoft/icms2#599]
В текущей ветке (2.8.0 и ранее) при смене шаблона все добавленные ранее (в другой шаблон, разумеется) виджеты становятся недоступны.

И приходится их не только расставлять по новой (что самое малое из бед), а по новой настраивать (выбирать тип контента, указывать класс, меню и т.п.).

Предложение. При смене шаблона копировать виджеты с предыдущего шаблона и помещать их в блок "Неиспользуемые виджеты". Визуально это можно сделать дополнительным окном в момент сохранения настроек, если шаблон изменился: "Вы сменили шаблон, скопировать виджеты? Да/Да, по возможности сохранить позиции/Нет"

Подробное обсуждение в теме http://www.instantcms.ru/forum/thread27905-1.html

**Comment by WebManG** [comment]
Функция действительно очень нужная. Но, на мой взгляд, делать её автоматической при переключении шаблона, да ещё и с всплывающим запросом - не удобно.
Лучше сделать кнопку на странице виджетов, по нажатию которой запрашивать источник, а потом либо копировать все виджеты в "Неиспользуемые", либо сначала пытаться подставлять их на одинаковые позиции, а оставшиеся - в "Неиспользуемые".
И копировать виджеты только для текущей страницы, а не для всей темы. Или при запросе источника давать админу выбрать: все страницы или только текущая.

**Comment by lakomet** [comment]
Поддерживаю, очень нужный функционал. Иногда приходится копировать виджеты, не только при смене шаблона, но и раставить несколько одинаковых виджетов, на разные страницы. Может можно, на вкладке "Виджеты и страницы", сделать ещё одно окно, такое же, как и "Неиспользуемые виджеты", но только "Копии виджетов"?

**Comment by WebManG** [comment]
Да, копирование отдельных настроенных виджетов - тоже удобная и нужная фича.
